File: algos/sorting/BubbleSort/JS/BubbleSort.js
function bubbleSort(n) {
    let donePass = false;
    let j = 0;
    
    while(true) {
        for(let i = 0; i < n.length - 1 - j; i++) {
            if(n[i] > n[i+1]) {
                let temp = n[i+1];
                n[i+1]  = n[i];
                n[i] = temp;
                donePass = true;
            }
        }
        if(donePass == false) return n;
        else {
            j++;
            donePass = false;
        }
    }
}

File: algos/sorting/BubbleSort/Java/BubbleSort.java
package com.dejanvuk.algos.sorting.BubbleSort.Java;

public class BubbleSort {

	public static void main(String[] args) {	
		int arr[] = {5,4,3,2,1};
		BubbleSort bs = new BubbleSort();
		bs.sort(arr);	
	}
	
	public void sort(int[] array) {
		boolean swapped;
		
		for(int i = 0 ; i < array.length - 1; i++) {
			swapped = false;
			for(int j = 0; j < array.length - i - 1; j++) {
				if(array[j] > array[j+1]) {
					int temp = array[j+1];
					array[j+1] = array[j];
					array[j] = temp;
					swapped = true;
				}
			}
			if(!swapped) return;
		}
	}
}
